Apeswap-Bots/detect-large-banana-token-mints/src/agent.spec.ts

```diff
@@ -0,0 +1,80 @@
+import { Finding, HandleTransaction, TransactionEvent } from "forta-agent";
+import { ethers } from "ethers";
+import { parseEther } from "@ethersproject/units";
+import { TestTransactionEvent } from "forta-agent-tools/lib/tests";
+import { contractMetaData, createFinding } from "./utils";
+import { mintTxHandler } from "./agent";
+import { GLOBALS, FUNCTION_ABI } from "./constants";
+const { BANANA_MINT_FUNCTION, BANANA_CONTRACT_ADDRESS_BNBCHAIN } = GLOBALS;
+
+describe("Detect Banana Token Mints", () => {
+  let handleTransaction: HandleTransaction;
+  let IBanana: ethers.utils.Interface;
+  let findings: Finding[];
+  let txEvent: TransactionEvent;
+
+  beforeAll(() => {
+    handleTransaction = mintTxHandler(BANANA_MINT_FUNCTION, contractMetaData);
+    IBanana = new ethers.utils.Interface(FUNCTION_ABI);
+  });
+
+  it("should return empty findings if there are no large banana mints", async () => {
+    txEvent = new TestTransactionEvent();
+    findings = await handleTransaction(txEvent);
+
+    expect(findings).toStrictEqual([]);
+    expect(findings.length).toStrictEqual(0);
+  });
+
+  it(`should return findings if ${BANANA_MINT_FUNCTION} is called on the contract`, async () => {
+    const parsedAmount = parseEther("25000");
+
+    txEvent = new TestTransactionEvent().addTraces({
+      to: BANANA_CONTRACT_ADDRESS_BNBCHAIN,
+      input: IBanana.encodeFunctionData("mint", [parsedAmount]),
+    });
+
+    findings = await handleTransaction(txEvent);
+
+    const [result] = findings;
+    const {
+      metadata: { from, to, value },
+    } = result;
+
+    expect(result).toStrictEqual(
+      createFinding({
+        from,
+        to,
+        value,
+      })
```

Comment:
This is not a valid test. You are using the info of the finding that the agent return to construct the expected finding so, you are not testing anything. You need to create the expected finding without making it dependent of the findings returned by the handler.
The same applies for other tests

Reply:
working to effect these suggestions